examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -191,5 +222,95 @@ bool NativeStackTrace::error_occured() const {
   return error_occurred;
 }
 
+bool NativeStackTrace::is_cached(const UnwindCache &map, const uint32_t &key) {
+  try {
+      map.at(key);
+      return true;
+  }
+  catch (const std::out_of_range&) {
+      logInfo(2, "No entry for %d in the cache\n", key);
+  }
+  return false;
+}
+
+UnwindCacheEntry NativeStackTrace::cache_get(const UnwindCache &map, const uint32_t &key) {
```

Comment:
Why not return a reference here?

Reply:
> @IzabellaRaulin , next times if you need to update from master, you can just merge it instead of rebasing. Rebase causes github to lose the current review state (I cannot see just the new changes). In any case we use squash-and-merge here, so whether you used merge or rebase to update your branch doesn't matter much eventually :)

Sorry, I didn't realize. But I put new changes in separate commits, so I hope that will help to track what is new.

Reply:
My objective was to limit `cache_get` to read-only access and to make it more explicit that the content of the cache is only changing when `cache_put` or `cache_delete_key` is called. The code is then easier to maintain especially since there is no need to have a reference there. 

Do you prefer having a reference here? Might you provide briefly why you think so? Maybe I missed something...I am open to your suggestion and looking forward to your opinion and guidelines.